### PR TITLE
Metric calc via helpers

### PR DESCRIPTION
**Pull request by sschneid**
This PR moves common metrics calculations to helpers.

### CODE REVIEW

File: lib/vmpooler/api/v1.rb [hunk 1]
```diff
@@ -5,48 +5,59 @@ class V1 < Sinatra::Base
...
+            total: 0
           },
           count: {
             total: 0
           }
-        },
-        queue: {
+        }
+
+        clone[:count][:total] = $redis.hlen('vmpooler__clone__' + date).to_i
```
Comment (colinPL) [inline lib/vmpooler/api/v1.rb]: Is `date` a date string or an actual date? If the latter, this needs `.to_s`. If the former, then I suggest renaming the variable to `date_str`. (methods below use "date" and treat it as a date object)

Comment (sschneid) [inline lib/vmpooler/api/v1.rb]: Anything that uses one of the helper functions should be passing the `date` param as a string.

File: lib/vmpooler/api/v1.rb [hunk 2]
```diff
@@ -5,48 +5,59 @@ class V1 < Sinatra::Base
...
+          clone[:duration][:total] = clone_times.reduce(:+).to_f
+          clone[:duration][:average] = (clone[:duration][:total] / clone[:count][:total]).round(1)
+          clone[:duration][:min], clone[:duration][:max] = clone_times.minmax
+        end
+
+        clone
+      end
+
+      def get_clone_times(date)
+        $redis.hvals('vmpooler__clone__' + date.to_s).map(&:to_f)
```
Comment (colinPL) [inline lib/vmpooler/api/v1.rb]: `date` is already a string.

File: lib/vmpooler/api/v1.rb (hunk 1 above)
Comment (sschneid) [inline lib/vmpooler/api/v1.rb]: (Updated anyway)

File: lib/vmpooler/api/v1.rb [hunk 3]
```diff
@@ -117,109 +148,56 @@ def mean(list)
...
-          # this prevents divide by 0 problems.
-          if me[:clone][:count][:total] > 0
-            me[:clone][:duration][:average] = mean(clone_times)
-          else
-            me[:clone][:duration][:average] = 0
-          end
-        end
+      result[:daily].each do |daily|
+        daily_clone_counts.push(daily[:clone][:count][:total])
+        daily_clone_durations.push(daily[:clone][:duration][:min])
```
Comment (colinPL) [inline lib/vmpooler/api/v1.rb]: If this `daily` did not clone anything (`[:count][:total] == 0`), then `min` and `max` will be the default of `0`. Appending those values to `daily_clone_durations` will cause future processing to select `0` as the overall minimum duration for this summary. These values should only be appended if the `daily` cloned something.

**Comment by colinPL** [comment]
Once the min/max comment is addressed, I am :+1:

File: lib/vmpooler/api/v1.rb (hunk 3 above)
Comment (sschneid) [inline lib/vmpooler/api/v1.rb]: Addressed in https://github.com/sschneid/vmpooler/commit/e347e2b6b4337d78dfc9dab908e09584a0bb3c5a.

**Pull request merged**